lib/servers/stepper_motor_server/stepper_motor_server.py

```diff
@@ -0,0 +1,132 @@
+# Created on Feb 12, 2017
+# @author: Anthony Ransford
+
+"""
+### BEGIN NODE INFO
+[info]
+name = stepper_motor_server
+version = 1.0
+description = drives stepper motors
+instancename = stepper_motor_server
+[startup]
+cmdline = %PYTHON% %FILE%
+timeout = 20
+[shutdown]
+message = 987654321
+timeout = 20
+### END NODE INFO
+"""
+
+from labrad.types import Value
+from labrad.devices import DeviceServer, DeviceWrapper
+from labrad.server import setting
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+TIMEOUT = Value(5, 's')  # serial read timeout
+BAUDRATE = 115200
+
+
+class stepperController(DeviceWrapper):
+
+    """
+    arduino dual stepper devices
+    """
+
+    @inlineCallbacks
+    def connect(self, server, port):
+        """Here we make a connection to the serial server in LabRAD where all
+        the serial communication is handled"""
+        print 'connecting to "%s" on port "%s"...' % (server.name, port),
+        self.server = server
+        self.ctx = server.context()  # grabs an identification number from the server
+        self.port = port
+        # The following opens a communication on a com port specifying a baudrate and a timeout
+        p = self.packet()
+        p.open(port)
+        p.baudrate(BAUDRATE)
+        p.read()  # clear out the read buffer
+        p.timeout(TIMEOUT)
+        yield p.send()
+
+    def packet(self):
```

Comment:
I think this all should be part of the DeviceWrapper but for whatever reason this is not true. Perhaps someone should update labrad.devices and submit a PR